sdk/communication/communication-jobrouter/src/generated/src/operations/jobRouter.ts

```diff
@@ -0,0 +1,827 @@
+ */
+
+import { PagedAsyncIterableIterator } from "@azure/core-paging";
+import { JobRouter } from "../operationsInterfaces";
+import * as coreClient from "@azure/core-client";
+import * as Mappers from "../models/mappers";
+import * as Parameters from "../models/parameters";
+import { JobRouterApiClient } from "../jobRouterApiClient";
+import {
+  RouterJobItem,
+  JobRouterListJobsNextOptionalParams,
+  JobRouterListJobsOptionalParams,
+  RouterWorkerItem,
+  JobRouterListWorkersNextOptionalParams,
+  JobRouterListWorkersOptionalParams,
+  RouterJob,
+  JobRouterUpsertJobOptionalParams,
+  JobRouterUpsertJobResponse,
+  JobRouterGetJobOptionalParams,
+  JobRouterGetJobResponse,
+  JobRouterDeleteJobOptionalParams,
+  JobRouterReclassifyJobActionOptionalParams,
+  JobRouterReclassifyJobActionResponse,
+  JobRouterCancelJobActionOptionalParams,
+  JobRouterCancelJobActionResponse,
+  JobRouterCompleteJobActionOptionalParams,
+  JobRouterCompleteJobActionResponse,
+  JobRouterCloseJobActionOptionalParams,
+  JobRouterCloseJobActionResponse,
+  JobRouterListJobsResponse,
+  JobRouterGetInQueuePositionOptionalParams,
+  JobRouterGetInQueuePositionResponse,
+  JobRouterUnassignJobActionOptionalParams,
+  JobRouterUnassignJobActionResponse,
+  JobRouterAcceptJobActionOptionalParams,
+  JobRouterAcceptJobActionResponse,
+  JobRouterDeclineJobActionOptionalParams,
+  JobRouterDeclineJobActionResponse,
+  JobRouterGetQueueStatisticsOptionalParams,
+  JobRouterGetQueueStatisticsResponse,
+  RouterWorker,
+  JobRouterUpsertWorkerOptionalParams,
+  JobRouterUpsertWorkerResponse,
+  JobRouterGetWorkerOptionalParams,
+  JobRouterGetWorkerResponse,
+  JobRouterDeleteWorkerOptionalParams,
+  JobRouterListWorkersResponse,
+  JobRouterListJobsNextResponse,
+  JobRouterListWorkersNextResponse
+} from "../models";
+
+/// <reference lib="esnext.asynciterable" />
+/** Class containing JobRouter operations. */
+export class JobRouterImpl implements JobRouter {
+  private readonly client: JobRouterApiClient;
+
+  /**
+   * Initialize a new instance of the class JobRouter class.
+   * @param client Reference to the service client
+   */
+  constructor(client: JobRouterApiClient) {
+    this.client = client;
+  }
+
+  /**
+   * Retrieves list of jobs based on filter parameters
+   * @param options The options parameters.
+   */
+  public listJobs(
+    options?: JobRouterListJobsOptionalParams
+  ): PagedAsyncIterableIterator<RouterJobItem> {
+    const iter = this.listJobsPagingAll(options);
+    return {
+      next() {
+        return iter.next();
+      },
+      [Symbol.asyncIterator]() {
+        return this;
+      },
+      byPage: () => {
+        return this.listJobsPagingPage(options);
+      }
+    };
+  }
+
+  private async *listJobsPagingPage(
+    options?: JobRouterListJobsOptionalParams
+  ): AsyncIterableIterator<RouterJobItem[]> {
+    let result = await this._listJobs(options);
```

Comment:
You will need to enable tracing. Maybe you can add it as a follow up PR, but ideally should be in for beta. This is an example of how tracing is integrated. If you decide to make this in a separate PR please file an issue to track it and add it as a comment here

https://github.com/Azure/azure-sdk-for-js/blob/295dbb025f646c6e846da877abda9e9d6b2ff8f8/sdk/template/template/src/configurationClient.ts#L143